### ISSUE TITLE
[kubernetes] Remote function scheduled to worker without a GPU despite specyfying num_gpus=1

### ISSUE THREAD

**Issue by janpawlowskiof** [ray-project/ray#16877]
In the helm chart I specify `GPU: 0` for head node and `GPU: 1` for a worker node. 
Then a schedule a simple function decorated with `@ray.remote(num_gpus=1)` expecting it to be only scheduled to worker nodes when called (since head node has no gpu), but it is scheduled to head node as well as worker nodes.
Also I checked that the GPU is actaully available only on the worker nodes as it should be.

This is not new to version 1.4.1

Tested version:
Ray version `1.4.1` with python `3.7`
Dockerhub image: `rayproject/ray-ml:1.4.1`

Take a look at reproduction.ipynb notebook.
Helm chart that I used (which is basically the one from an example in documentation) is included as well.
[reproduction.zip](https://github.com/ray-project/ray/files/6765587/reproduction.zip)

For me the output looks like this:
![obraz](https://user-images.githubusercontent.com/17483828/124501973-d57c0200-ddc2-11eb-8529-2810f7f0b1ea.png)

This runs on a configurated local kubernetes cluster, so obviously I can do only so much when it comes to testing on clean environment.

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

I will appreciate any help with this bug.

**Comment by richardliaw** [comment]
cc @DmitriGekhtman @AmeerHajAli

**Comment by DmitriGekhtman** [comment]
Thanks for pointing this out -- last time I looked at ResourceDemandSchedulet code it looked like it was overriding user settings with resources detected at runtime -- looks from this issue that that's indeed the case. This is a general autoscaler problem (not k8s specific)

**Comment by richardliaw** [comment]
@DmitriGekhtman got it, Is it possible to fix this for 1.5.0?

**Comment by DmitriGekhtman** [comment]
@richardliaw most likely

@janpawlowskiof could you provide more details about your kubernetes setup?

What could help in particular: details about the Ray head pod at runtime and about the k8s node the head pod is running on.
What's the k8s node's gpu setup?

**Comment by DmitriGekhtman** [comment]
Might have misdiagnosed the cause -- some details about the K8s nodes could help.

@wuisawesome just curious -- if a Ray node is started with a 0 GPU override, will GPU be absent from [resources_total](https://github.com/ray-project/ray/blob/500248163f88ce7a9a3a7b7c6fcb55e83dd27a8e/src/ray/protobuf/gcs.proto#L451)

**Comment by DmitriGekhtman** [comment]
Also @janpawlowskiof if things were working properly, this wouldn't be necessary, but I think you could get the correct behavior by adding `rayResources: {"GPU": 0}` to the head config.

I'm guessing the Ray head pod is scheduled on a gpu k8s node and Ray's gpu auto-detection is erroneously picking up the node's gpu as available.

**Comment by janpawlowskiof** [comment]
Adding `rayResources: {"GPU": 0}` does fix this issue! Thanks a bunch!

Although I don't think that what your guess is correct. 
The the head node really does not have any gpus according to `tf.config.experimental.list_physical_devices('GPU')`

**Comment by janpawlowskiof** [comment]
For all my prior tests I used a kubernetes cluster configured with [this](https://github.com/NVIDIA/k8s-device-plugin#preparing-your-gpu-nodes) guide for a gpu support.

Docker image I used was `rayproject/ray-ml:1.4.0-py37-gpu`.

The only unusual thing (that I don't know if it's is supposed to happen?) was that I had to run these commands in the docker image in order for tensorflow to recognize the GPU (RTX 2080Ti).

```
ENV LD_LIBRARY_PATH=/usr/local/cuda-11.0/targets/x86_64-linux/lib
RUN sudo ln -s /usr/local/cuda-11.0/targets/x86_64-linux/lib/libcusolver.so.10 /usr/local/cuda-11.0/targets/x86_64-linux/lib/libcusolver.so.11
```

What information I should provide?
